docs/src/manual/nlp.md

```diff
@@ -741,4 +742,29 @@ Subject to
  (f_1(x) + f_2(x)) - 1.0 ≤ 0
 ```
 
+## Known performance issues
+
+The macro-based input to JuMP's nonlinear interface can cause a performance
+issue if you:
+
+ 1. write a macro with a large number (hundreds) of terms
+ 2. call that macro from within a function instead of from the top-level in
+    global scope.
+
+The first issue does not depend on the number of resulting terms in the
+mathematical expression, but rather the number of terms in the Julia `Expr`
+representation of that expression. For example, the expression
+`sum(x[i] for i in 1:1_000_000)` contains one million mathematical terms, but
+the `Expr` representation is just a single sum.
+
+The most common cause, other than a lot of tedious typing, is if you write a
```

Comment:
I would suggest rephrasing `The most common cause` as `The second cause`

Reply:
Hmm. This might be my NZ English coming through. I meant it as written: that the most common cause is transpilation, because no one can really trigger this without a lot of tedious typing.

Reply:
I see, that makes more sense indeed. Let's keep the original formulation then :+1: